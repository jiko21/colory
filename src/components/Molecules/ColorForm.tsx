import { TextField, View } from '@adobe/react-spectrum';
import React from 'react';

interface Props {
  color: string;
  handleChange: (value: string) => void;
}

const ColorForm: React.FC<Props> = ({ color, handleChange }) => {
  return (
    <>
      <View>
        <input type="color" value={color} onChange={(e) => handleChange(e.target.value)} data-testid="color-picker" />
      </View>
      <View>
        <TextField value={color} onChange={handleChange} data-testid="color-text" />
      </View>
    </>
  );
};

export default ColorForm;
