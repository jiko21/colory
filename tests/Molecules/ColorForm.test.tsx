import ColorForm from '@/components/Molecules/ColorForm';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

describe('components/Molecules/ColorForm.tsx', () => {
  it('correctly call handleChange with color picker', () => {
    const color = '#FF0000';
    const handleChangeMock = jest.fn();
    render(<ColorForm color={color} handleChange={handleChangeMock} />);
    fireEvent.change(screen.getByTestId('color-picker'), {
      target: { value: '#000000' },
    });
    expect(handleChangeMock).toBeCalledWith('#000000');
  });

  it('correctly call handleChange with input text', () => {
    const color = '#FF0000';
    const handleChangeMock = jest.fn();
    render(<ColorForm color={color} handleChange={handleChangeMock} />);
    fireEvent.change(screen.getByTestId('color-text'), {
      target: { value: '#000000' },
    });
    expect(handleChangeMock).toBeCalledWith('#000000');
  });
});
