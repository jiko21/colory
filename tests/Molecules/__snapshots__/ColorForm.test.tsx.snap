// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Molecules/ColorForm.tsx correctly render 1`] = `
<div>
  <div>
    <input
      data-testid="color-picker"
      type="color"
      value="#FF0000"
    />
  </div>
  <div>
    <div
      class="_spectrum-Textfield_61339"
    >
      <input
        class="_spectrum-Textfield-input_61339"
        data-testid="color-text"
        id="react-aria-2737256666-1"
        type="text"
        value="#FF0000"
      />
    </div>
  </div>
</div>
`;
